### PR TITLE
Lagom: Remove GLOB patterns for most libraries, and build using their CMakeLists in Userland

### PR DESCRIPTION
**Pull request by ADKaster**
Bunch of cleanup here:

- Create an AK_SOURCES variable for use n places that want to include AK in their library (LibC, DynamicLodaer, Lagom LibCore)
- Convert lagom_lib rules to use add_serenity_subdirectory helper, which will defer to the CMakeLists.txt in each library or application or service's individual subdirectory.
- Remove logic for Lagom vs Serenity build directory in code generator helpers
- Leave some low hanging fruit FIXMEs for a crafty CMake apprentice to pick up if they want.

### CODE REVIEW

File: Meta/Lagom/CMakeLists.txt [hunk 1]
```diff
@@ -325,274 +324,42 @@ endif()
...
+    set(lagom_standard_libraries
+        Archive
+        Audio
+        Compress
+        Crypto
+        DNS
+        Gemini
+        Gfx
+        GPU
+        GL
```
Comment (linusg) [inline Meta/Lagom/CMakeLists.txt:336]: `swap()`

File: Meta/Lagom/CMakeLists.txt [hunk 2]
```diff
@@ -325,274 +324,42 @@ endif()
...
+        IPC
+        JS
+        Line
+        Locale
+        Markdown
+        PDF
+        Regex
+        SoftGPU
+        Syntax
+        SQL
```
Comment (linusg) [inline Meta/Lagom/CMakeLists.txt:348]: `swap()`

File: Meta/Lagom/CMakeLists.txt [hunk 3]
```diff
@@ -607,23 +374,48 @@ if (BUILD_LAGOM)
...
+    endforeach()
+
+    # GUI
+    set(LIBGUI_SOURCES
+        ModelIndex.cpp
+        Model.cpp
+        Icon.cpp
+        GML/Lexer.cpp
+        GML/Parser.cpp
+        GML/SyntaxHighlighter.cpp
```
Comment (linusg) [inline Meta/Lagom/CMakeLists.txt:401]: Um

File: Userland/Libraries/LibCore/CMakeLists.txt [hunk 4]
```diff
@@ -38,8 +34,14 @@ set(SOURCES
...
     Version.cpp
 )
-if (NOT ANDROID)
-    list(APPEND SOURCES Account.cpp)
+if (NOT ANDROID AND NOT WIN32)
+    list(APPEND SOURCES
+        Account.cpp
+        FilePermissionsMask.cpp
+        Group.cpp
+        GetPassword.cpp
```
Comment (linusg) [inline Userland/Libraries/LibCore/CMakeLists.txt:42]: `swap()`

**Review by linusg — CHANGES_REQUESTED** [review]

File: AK/CMakeLists.txt [hunk 5]
```diff
@@ -1,2 +1,34 @@
...
+    kmalloc.cpp
+    FlyString.cpp
+    Time.cpp
+    Random.cpp
+    Assertions.cpp
+    JsonPath.cpp
+    UUID.cpp
+    URLParser.cpp
+    Utf8View.cpp
+    StringUtils.cpp
```
[the remark this answers is not in the document]
Comment (ADKaster) [reply AK/CMakeLists.txt:26]: That's the order of inodes on my machine, I copy pasted from `find` :)

**Comment by nico** [comment]
Typo "varaible" in commit message

**Comment by ADKaster** [comment]
Re-ordered lists and fixed typos, thanks for the reviews!

**Review by linusg — APPROVED** [review]

**Pull request merged**